tests/Kernel/LanguageSwitcherTest.php

```diff
@@ -15,6 +16,65 @@ class LanguageSwitcherTest extends MultilingualAbstractKernelTestBase {
    * Test language switcher link list rendering.
    */
   public function testLanguageSwitcherLinkListRendering(): void {
+    $this->assertDefaultLanguageBlock();
+
+    $lang_config = $this->container->get('config.factory')->get('language.negotiation');
+
+    // Create the Icelandic language.
+    $language = ConfigurableLanguage::createFromLangcode('is');
+    $language->setThirdPartySetting('oe_multilingual', 'category', 'non_eu');
```

Comment:
Before you set the category, run an assertion that our defaults are still in place. We can have Icelanding but if not categorised, the defaults apply: only EU langs show, no category title, etc.